File: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
class Solution{   
public:
    long maximumSumSubarray(int k, vector<int> &Arr , int n){
        // code here 
        long long sum = 0;
        long long temp = 0;
        
        for(int i=0; i<k; i++){
            temp += Arr[i];
        }
        sum = temp;
        for(int i=k; i<n; i++){
            sum += Arr[i]-Arr[i-k];
            temp = max(temp,sum);
        }
        
        return temp;
    }
};

//{ Driver Code Starts.
int main() 
{ 
    int t;
    cin>>t;
    while(t--)
    {
        int N,K;
        cin >> N >> K;;
        vector<int>Arr;
        for(int i=0;i<N;++i){
            int x;
            cin>>x;
            Arr.push_back(x);
        }
        Solution ob;
        cout << ob.maximumSumSubarray(K,Arr,N) << endl;
    }
    return 0; 
} 
// } Driver Code Ends

File: Difficulty: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.java
class Solution {
    public int maxSubarraySum(int[] arr, int k) {
        // Code here
        int n = arr.length;
        int maxSum = 0;
        int windowSum = 0;
        
        for(int i=0; i<k; i++){
            windowSum += arr[i];
        }
        maxSum = windowSum;
        
        for(int j=k ; j<n; j++){
            windowSum += arr[j];
            windowSum -= arr[j-k];
            
            maxSum = Math.max(maxSum,windowSum);
        }
        return maxSum;
        
    }
}